File: 项目文件/backend/app/schemas/project_member.py
"""项目成员 Pydantic 模型"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProjectMemberCreate(BaseModel):
    """创建项目成员请求"""

    project_id: uuid.UUID
    user_id: uuid.UUID
    role_title: str | None = None
    notes: str | None = None
    is_owner: bool = False
    is_active: bool = True
    joined_at: datetime | None = None
    left_at: datetime | None = None


class ProjectMemberUpdate(BaseModel):
    """更新项目成员请求"""

    role_title: str | None = None
    notes: str | None = None
    is_owner: bool | None = None
    is_active: bool | None = None
    joined_at: datetime | None = None
    left_at: datetime | None = None


class ProjectMemberResponse(BaseModel):
    """项目成员响应"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    user_id: uuid.UUID
    role_title: str | None = None
    notes: str | None = None
    is_owner: bool
    is_active: bool
    joined_at: datetime | None = None
    left_at: datetime | None = None
    created_at: datetime


class ProjectMemberListResponse(BaseModel):
    """项目成员列表响应"""

    items: list[ProjectMemberResponse]
    total: int

File: 项目文件/frontend/src/types/project-member.ts
/** 项目成员类型定义 */

export interface ProjectMember {
  id: string;
  project_id: string;
  user_id: string;
  role_title: string | null;
  notes: string | null;
  is_owner: boolean;
  is_active: boolean;
  joined_at: string | null;
  left_at: string | null;
  created_at: string;
}

export interface ProjectMemberCreate {
  project_id: string;
  user_id: string;
  role_title?: string;
  notes?: string;
  is_owner?: boolean;
  is_active?: boolean;
  joined_at?: string;
  left_at?: string;
}

export interface ProjectMemberUpdate {
  role_title?: string;
  notes?: string;
  is_owner?: boolean;
  is_active?: boolean;
  joined_at?: string;
  left_at?: string | null;
}

export interface ProjectMemberListResponse {
  items: ProjectMember[];
  total: number;
}
